plugin/src/features/Compilation/index.tsx

```diff
@@ -146,19 +168,25 @@ export const Compilation = ({ setAccordian }: CompilationProps) => {
       const artifacts: string[] = []
 
       const contractsToAdd: Contract[] = []
-      for (const file of compileResult.file_content) {
-        if (!file.is_contract) continue
-        const contract = JSON.parse(file.file_content) as Contract
-        contractsToAdd.push(contract)
+      if (type === 'PROJECT') {
+        for (const file of compileResult.file_content) {
+          if (!file.is_contract || !file.file_name.startsWith('contracts/')) continue
+          const contract = JSON.parse(file.file_content) as Contract
+          contractsToAdd.push(contract)
+        }
+      } else {
+        for (const file of compileResult.file_content) {
+          if (!file.is_contract || !file.file_name.includes(currentFilename)) continue
```

Comment:
!file.file_name.includes(currentFilename) will also work for the case

/root
--/a
----Bank.sol
----/b
------Bank.sol

Reply:
This shall change `currentFilename` to `currentFilePath` and get `currentFilename` from `currentFilePath ` atom

Reply:
Yeah that's better.